### PR TITLE
Migrate VCFTrack from SVG→canvas

### LINKED ISSUE #255: D3/SVG → canvas
I experimented with switching from SVG to Canvas in `PileupTrack`. It:

1. is simpler to code against than D3
2. is dramatically faster (at least 10x)
3. works better with Flow (to which d3 is opaque)

I'm clearing the entire canvas and redrawing it on every frame and _still_ getting 10x performance, so there are even more wins to be had if we need them.

Visualizations to port:

- [x] PileupTrack
- [x] CoverageTrack
- [x] GeneTrack
- [x] GenomeTrack
- [x] LocationTrack
- [x] ScaleTrack
- [ ] VariantTrack

Other cleanup:

- [x] Add back support for strandedness in GeneTrack

The components don't interact, so these can be done one at a time. We can also leave some of the less performance-intensive tracks (e.g. `VariantTrack`) as SVG if we want.

We lose the ability to color/style tracks using CSS when we do this. I think that's probably worth it.

### PR DESCRIPTION
**Pull request by danvk**
Fixes #255 

I took the opportunity to factor out some code into a shared `canvas-utils` module.

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/pileup.js/277)

### CODE REVIEW

File: src/main/GenomeTrack.js [hunk 1]
```diff
@@ -27,7 +28,7 @@ var GenomeTrack = React.createClass({
     onRangeChange: React.PropTypes.func.isRequired,
   },
   render: function(): any {
-    return <div><canvas ref='canvas' /></div>;
+    return <canvas />;
   },
   componentDidMount: function() {
     var div = this.getDOMNode();
```
Comment (ihodes) [inline src/main/GenomeTrack.js]: Did not know you could do this, nice. (As an aside, are you using `div` here? (It's also `canvas` now (sorry for chiming in))).

Comment (danvk) [inline src/main/GenomeTrack.js]: You can in old-style React components (i.e. `React.createClass`) but not in new-style ES6 classes. With those you have to do `React.findDOMNode(this)`.

Comment (danvk) [inline src/main/GenomeTrack.js]: (also, it was dead code -- will remove)

**Comment by danvk** [comment]
This revealed another advantage of using canvas: previously it looked like the VariantTrack was being tested, but it really wasn't (the d3 methods were being called with empty arrays). With canvas, the drawing code clearly wasn't being run. So I added a test.

**Comment by armish** [comment]
just two minor comments -- but looks great! :+1:

Kudos for refactoring.

---

Reviewed 7 of 8 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

<sup>**[src/main/canvas-utils.js, line 2 \[r2\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/277#-K-5AbXns1ZvhtIm0l_0)** ([raw file](https://github.com/hammerlab/pileup.js/blob/1c183f4f2e38cb85a7e97d2a34dfc55c6571e7f2/src/main/canvas-utils.js#L2)):</sup>
Thanks for factoring these out!

---

<sup>**[src/main/VariantTrack.js, line 56 \[r2\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/277#-K-5BR-VzrkLNEgEQm5d)** ([raw file](https://github.com/hammerlab/pileup.js/blob/1c183f4f2e38cb85a7e97d2a34dfc55c6571e7f2/src/main/VariantTrack.js#L56)):</sup>
is it worth refactoring this into: `{width, height} = this.props` to make it consistent across tracks?

---

<sup>**[src/main/VariantTrack.js, line 103 \[r2\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/277#-K-5BjLexXSXFIt4XdKe)** ([raw file](https://github.com/hammerlab/pileup.js/blob/1c183f4f2e38cb85a7e97d2a34dfc55c6571e7f2/src/main/VariantTrack.js#L103)):</sup>
I know this is unrelated, but maybe it is better if we `console.log` or `console.dir` this one. IMHO alert behavior is pretty annoying especially with lines and lines of JSON in it ;)

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/pileup.js/277)

**Comment by danvk** [comment]
Thanks for the review -- will merge on green.

---

Review status: 9 of 10 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

<sup>**[src/main/VariantTrack.js, line 56 \[r2\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/277#-K-5BR-VzrkLNEgEQm5d)** ([raw file](https://github.com/hammerlab/pileup.js/blob/1c183f4f2e38cb85a7e97d2a34dfc55c6571e7f2/src/main/VariantTrack.js#L56)):</sup>
Done.

---

<sup>**[src/main/VariantTrack.js, line 103 \[r2\]](https://reviewable.io:443/reviews/hammerlab/pileup.js/277#-K-5BjLexXSXFIt4XdKe)** ([raw file](https://github.com/hammerlab/pileup.js/blob/1c183f4f2e38cb85a7e97d2a34dfc55c6571e7f2/src/main/VariantTrack.js#L103)):</sup>
Yeah, this needs to be rethought. See #273.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/pileup.js/277)

**Comment by armish** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/hammerlab/pileup.js/277)

**Pull request merged**